routes/signMessage.ts

```diff
@@ -1,19 +1,96 @@
-import { signMessage } from '../lib/signatures'
-import { app } from '../server'
+import { ed25519 } from '@noble/curves/ed25519'
+import { blake3 } from '@noble/hashes/blake3'
+import { base64 } from '@scure/base'
+import { app } from '../app'
+import { messageDataToUint8Array } from '../buffers/buffers'
+import { kms } from '../clients/aws'
+import { authDb } from '../database/watcher'
+import { selectKeysQuery, selectSessionQuery } from '../lib/queries'
+import { signMessageWithKey } from '../lib/signatures'
+import { lucia } from '../lucia/auth'
 
-app.post('/sign', async (c) => {
+import type { Message } from '../utils/types'
+import { isMessage } from '../utils/types'
+
+app.post('/signMessage', async (c) => {
   try {
-    const { message } = await c.req.json()
+    const { sessionId, message } = await c.req.json()
+
+    if (!sessionId || !message) {
+      return c.json({ success: false, message: 'Missing parameters' }, 400)
+    }
 
-    if (!message) {
-      return c.json({ success: false, message: 'No message provided' }, 400)
+    if (!isMessage(message)) {
+      return c.json({ success: false, message: 'Invalid message format' }, 400)
     }
 
-    const signedMessage = signMessage(message)
+    const { session, user } = await lucia.validateSession(sessionId)
+
+    if (!session) {
+      return c.json({ success: false, message: 'Invalid session' }, 404)
+    }
+
+    const sessionResult = await authDb.query(selectSessionQuery, [sessionId])
+    const { userid } = sessionResult.rows[0]
+
+    if (userid.toString() !== message.messageData.rid) {
+      return c.json({ success: false, message: 'RID mismatch' }, 400)
+    }
+
+    const keysResult = await authDb.query(selectKeysQuery, [userid])
+
+    if (keysResult.rows.length === 0) {
+      return c.json({ success: false, message: 'Keys not found' }, 404)
+    }
+
+    const { encryptedprivatekey, publickey } = keysResult.rows[0]
+
+    console.log('HRE', publickey)
+
+    const computedHash = blake3(messageDataToUint8Array(message.messageData))
+
+    const computedHashBase64 = base64.encode(computedHash)
+    console.log({ computedHashBase64 })
+    console.log({ messagehash: message.hash })
+
+    if (computedHashBase64 !== message.hash) {
+      return c.json({ success: false, message: 'Invalid message hash' }, 400)
+    }
+
+    // Decrypt the private key using AWS KMS
+    const decryptedPrivateKey = await kms
+      .decrypt({
+        CiphertextBlob: Buffer.from(encryptedprivatekey, 'base64'),
```

Comment:
u use Buffer.from notation a lot. is it actually cuz u need to be turning all these things to buffer? i rarely use it so am confused why necessary or preferable over a diff type of helper